Content.Server/Pinpointer/PinpointerSystem.cs

```diff
@@ -127,14 +200,47 @@ public override void Update(float frameTime)
         return l.Count > 0 ? l.First().Value : null;
     }
 
+    /// <summary>
+    ///     Try to find the closest entity from stealGroup on a current map
+    ///     Will return null if can't find anything
+    /// </summary>
+    private EntityUid? FindTargetFromStealGroup(EntityUid uid, string group)
+    {
+        foreach (var target in _lookup.GetEntitiesInRange<StealTargetComponent>(_transform.GetMapCoordinates(uid), 100))
```

Comment:
Don't make unconstrained entity searches. This gets every single entity within your range (which will be alot). Use a component bounded search instead since this only searches entities that have the specified component